### PR TITLE
PDA-118: Support groups on task list endpoint

### PR DESCRIPTION
**Pull request by werbth**

### CODE REVIEW

**Comment by entando-service** [comment]
Something went wrong. Investigate!

File: src/main/java/org/entando/plugins/pda/pam/service/task/model/KieTaskSummaryResponse.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import lombok.Setter;
+import org.entando.plugins.pda.pam.service.task.KieDateDeserializer;
+import org.kie.server.api.model.instance.TaskSummary;
+import org.springframework.beans.BeanUtils;
+
+@Getter
+@Setter
+@SuppressFBWarnings({"EI_EXPOSE_REP", "EI_EXPOSE_REP2"})
+@SuppressWarnings("PMD.TooManyFields")
+public class KieTaskSummaryResponse {
```
Comment (ffleandro) [inline src/main/java/org/entando/plugins/pda/pam/service/task/model/KieTaskSummaryResponse.java:17]: You can use TaskSummary from the KIE Client library, it's the exact same object

**Review by ffleandro — DISMISSED** [review]
Have you considered creating a PR on the KIE Client Library (or forking the library to Entando repository)?
It would generate way less code and we could keep the same standards as all other endpoints.

File: src/main/java/org/entando/plugins/pda/pam/service/task/model/KieTaskSummaryResponse.java (hunk 1 above)
Comment (werbth) [reply src/main/java/org/entando/plugins/pda/pam/service/task/model/KieTaskSummaryResponse.java:17]: I needed to add the JsonProperty annotations, like this one `@JsonProperty("task-id")`. On their side they use a different parser.

**Comment by werbth** [comment]
> Have you considered creating a PR on the KIE Client Library (or forking the library to Entando repository)?
> It would generate way less code and we could keep the same standards as all other endpoints.

We can do that and as already discussed there is a ticket on their side, but this task shouldn't wait for that. When it is integrated in their client library we can change the implementation to use their client.

**Comment by werbth** [comment]
Retest this please

**Comment by entando-service** [comment]
All is well

**Review by BrenoQVDS — APPROVED** [review]

**Pull request merged**